### PR TITLE
Rename library functions. Remove unused ones

### PR DESCRIPTION
**Pull request by Arcticae**
Cairo PR: [68](https://github.com/software-mansion-labs/cairo/pull/68)

This PR also removes unused and obsolete libfuncs versions for cairo0

### Checklist
- [x] Relevant issue is linked
- [x] Docs updated/issue for docs created
- [x] Added relevant tests

### CODE REVIEW

File: protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py [hunk 1]
```diff
@@ -21,12 +21,12 @@ def __init__(self, contracts_controller: ContractsController):
 
     @property
     def name(self) -> str:
-        return "deploy_tp"
+        return "deploy_impl"
 
     def _build(self) -> Callable[[list[int], int, int], Any]:
-        return self.deploy_prepared_tp
+        return self.deploy_prepared_impl
```
Comment (karol-bisztyga) [inline protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py:27]:
```suggestion
        return self.deploy
```

File: protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py [hunk 2]
```diff
@@ -21,12 +21,12 @@ def __init__(self, contracts_controller: ContractsController):
...
 
-    def deploy_prepared_tp(
+    def deploy_prepared_impl(
```
Comment (karol-bisztyga) [inline protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py:29]:
```suggestion
    def deploy(
```

File: scripts/install_cairo_bindings.sh [hunk 3]
```diff
@@ -13,7 +13,6 @@ if [ "$1" == "--cleanup" ]; then
 fi
 
 function install_dev() {
-  git pull --recurse-submodules
```
Comment (karol-bisztyga) [inline scripts/install_cairo_bindings.sh:16]: can you explain this change and how it is related?

File: scripts/install_cairo_bindings.sh [hunk 4]
```diff
@@ -24,7 +23,6 @@ function install_dev() {
 }
 
 function install_prod() {
-  git pull --recurse-submodules
```
Comment (karol-bisztyga) [inline scripts/install_cairo_bindings.sh:27]: can you explain this change and how it is related?

Comment (Arcticae) [reply scripts/install_cairo_bindings.sh]: [Revert unwanted change, delete intermediate function and rename deplo…](https://github.com/software-mansion/protostar/pull/1845/commits/61d9b414d84f482849a114f029114024adb8c1ee)

File: scripts/install_cairo_bindings.sh (hunk 3 above)
Comment (Arcticae) [reply scripts/install_cairo_bindings.sh]: [Revert unwanted change, delete intermediate function and rename deplo…](https://github.com/software-mansion/protostar/pull/1845/commits/61d9b414d84f482849a114f029114024adb8c1ee)

File: protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py (hunk 2 above)
Comment (Arcticae) [reply protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py]: [Revert unwanted change, delete intermediate function and rename deplo…](https://github.com/software-mansion/protostar/pull/1845/commits/61d9b414d84f482849a114f029114024adb8c1ee)

File: protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py (hunk 1 above)
Comment (Arcticae) [reply protostar/cheatable_starknet/callable_hint_locals/deploy_hint_local.py]: [Revert unwanted change, delete intermediate function and rename deplo…](https://github.com/software-mansion/protostar/pull/1845/commits/61d9b414d84f482849a114f029114024adb8c1ee)

**Review by karol-bisztyga — APPROVED** [review]

**Review by piotmag769 — COMMENTED** [review]
Thanks for cleaning the mess I left in cairo fork xD

**Pull request merged**